add document elaborating on how to use TLS and examples (#391)
kubernetes-sigs/service-apis: docs-src/concepts.md

# API Concepts

> This document is a deep dive into the reasoning and design for the API.

Reviewer: s/the API/Service APIs/

## TLS Configuration

TLS is configured on Gateway listeners. Additionally, TLS certificates

> can be configured on Route objects for certain self-service use cases.

Reviewer 2: ```suggestion
can be configured on route objects for certain self-service use cases.
```

> TLS configuration is tied to Gateway listeners. Additionally, TLS certificates

Reviewer: COnsider the following minor change:

s/TLS configuration is tied to Gateway listeners./TLS is configured on Gateway listeners./

> can be configured at the Route-level for certain self-service use cases.

Reviewer 3: ```suggestion
can be configured on Route objects for certain self-service use cases.
```

> Please refer to the [TLS document](tls.md) for a deep dive on TLS.

Reviewer: Instead of TLS document, maybe the document should be called TLS Details, etc..
